### DISCUSSION TITLE
BRouter on Android 13

### DISCUSSION THREAD

**Discussion by DolomitesTom** [devemux86/cruiser#4] [Q&A] [answered]
BRouter doesn't work any more on Android 13. How can I then use BROUTER offline routing? I tried to reinstall Brouter, but I can't download any map because the selection of maps doesn't work. No map can be marked for download.

Additional Question: does the Graphopper offline routing still have the restriction not to be able to route over country borders?

**Comment by devemux86** [comment]
> Additional Question: does the Graphopper offline routing still have the restriction not to be able to route over country borders?

GraphHopper (offline) can provide offline routing for 1 activated region at a time.
Usually divided into countries, however it provides speed limits and road names.

**Comment by devemux86** [comment, marked as answer]
I tried it again with a clean install of BRouter on Android 13 and it works properly.

What storage have you selected?
Have you tried to restart the phone?

Please see below the steps I used on Android 13:
![BRouter1](https://user-images.githubusercontent.com/3484020/209925095-486c7eda-c244-4427-9369-f4bf7a7902d5.jpg)
![BRouter2](https://user-images.githubusercontent.com/3484020/209925100-178e0319-8587-40ad-94f3-33083e4a6ee8.jpg)
![BRouter3](https://user-images.githubusercontent.com/3484020/209925106-ba477056-5671-418b-a7f5-bc4d9307711c.jpg)
![BRouter4](https://user-images.githubusercontent.com/3484020/209925110-c431085a-8559-4fe4-bb18-493b4f47c6e7.jpg)
![BRouter5](https://user-images.githubusercontent.com/3484020/209925114-ccff37f8-b10c-4255-a747-07f3f431e2c5.jpg)

**Comment by DolomitesTom** [comment]
Thanks for the answers. I have used the external storage. I think I did reboot, but don't know it for sure any more. What I did in the meantime: I have zipped the file structure and copied from the one phone to the other and then BROUTER could be used again. It showed the saved maps and I could also configure it again. 

Speed Limits and road names (do you mean in the voice guidance?) are not so important for me. I live in the 3 countries area Germany, France, Luxemburg and Belgium is also not far away. Therefore I often cross borders and BRouter is better for me. But I recognized that then the settings in Cruiser don't play a role. If I set to quickest route or shortest, always what is set up in Brouter is taken. I guess that is normal, since Cruiser cannot overwrite this behaviour, right?

**Comment by devemux86** [comment]
> If I set to quickest route or shortest, always what is set up in Brouter is taken. I guess that is normal, since Cruiser cannot overwrite this behaviour, right?

Cruiser sends the Travel mode [Car, Bike, Foot] and Route type [Fastest, Shortest] to BRouter (offline or online).

Then the BRouter profiles that used are:

- Cruiser **Car** / **Fastest** = BRouter **car-fast**
- Cruiser **Car** / **Shortest** = BRouter **car-eco**
- Cruiser **Bike** / **Fastest** = BRouter **fastbike**
- Cruiser **Bike** / **Shortest** = BRouter **trekking**
- Cruiser **Foot** = BRouter **shortest**

You can find [here](https://brouter.de/brouter/profiles2/) all BRouter profiles ([documentation](https://brouter.de/brouter/costfunctions.html)).

**Comment by devemux86** [comment]
BRouter profiles are located on the phone at: `/Android/media/btools.routingapp/brouter/profiles2/`

Even in Android 13 you can modify them with any file manager.
That's BRouter's power and flexibility.

**Comment by DolomitesTom** [comment]
Thanks for explanation. Nevertheless I find it a bit complicated. I don't understand for example which settings trigger what.

Confusing for example is also that in this combination "Cruiser Bike / Fastest = BRouter fastbike" it takes german freeways, but with a bike it is not allowed to use freeways. Would it be possible to have different section profiles, for example from wp1 to wp2 fastest and from wp2 to wp3 shortest like in Kurviger 2? When using this option, the only option is "straigth line".

Have just downloaded 2 new maps, Belgium and Alps. Now the fuel stations are visible at zoom rate 10 after having deactivated and activated the map again. That's cool.

**Comment by devemux86** [comment]
> Confusing for example is also that in this combination "Cruiser Bike / Fastest = BRouter fastbike" it takes german freeways, but with a bike it is not allowed to use freeways.

Can you also check with the BRouter website?
(select profile at top-left, customize at top-right)
BRouter website: https://brouter.de/brouter-web

Also can you ask the BRouter authors with specific route examples?
They are more experienced, as they know much better their engine.

BRouter repository: https://github.com/abrensch/brouter
BRouter forum: https://groups.google.com/g/osm-android-bikerouting

> Would it be possible to have different section profiles, for example from wp1 to wp2 fastest and from wp2 to wp3 shortest like in Kurviger 2? When using this option, the only option is "straight line".

In a future version, I plan to add multiple profiles, regardless what the routers have.
So we could set "Straight line" (or other) in route planning with any routing provider.

> Have just downloaded 2 new maps, Belgium and Alps. Now the fuel stations are visible at zoom rate 10 after having deactivated and activated the map again. That's cool.

:+1:

**Comment by DolomitesTom** [comment]
>Can you also check with the BRouter website?

I tested a bit. There is in principle no difference between car fast and car eco. Even if I change car eco to assign "avoid_motorways     = true", it uses the german freeway. I would have expected to not use the freeway then.

In Brouter (Web) when using the "shortest.brf" then it really avoids the freeway and uses normal roads (at least at my tested section) which would then also be usable for motorcycles (theoretically). Since this profile doesn't contain "# Car profile assign validForCars        = true" I don't know if it would use MTB ways.
